### PR TITLE
Fix breaking laminas code change

### LINKED ISSUE #334: Undefined type 'Laminas\Code\Generator\Exception\ClassNotFoundException'.intelephense (1009)
Hi.
I have problems in time of wizard work.
I tried to install and generate classes, like here https://github.com/phpro/soap-client.

My composer.json:
```
{
    "name": "evgeny/public_html",
    "require": {
        "phpro/soap-client": "^1.3"
    },
    "require-dev": {
        "laminas/laminas-code": "^3.1.0"
    },
    "autoload": {
        "psr-4": {
            "App\\" : "."
        }
    }
}
```
What I did http://i.imgur.com/5nU3Ogs.png
WSDL, that I used http://webservices.gama-system.com/exchangerates.asmx?WSDL 
What I found in code http://i.imgur.com/4Vmds8J.png

What do I do wrong?
Thx.

### PR DESCRIPTION
**Pull request by veewee**
Fixes #334

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
-    level: 4
+    level: 5
     ignoreErrors:
         # This is something we hope to fix in PHP 7.4
         # https://wiki.php.net/rfc/covariant-returns-and-contravariant-parameters
         - '#Call to an undefined method Phpro\\.*(Context|RuleSet)Interface.*$#'
         # Symfony BC code makes phpstan cry:
         - '#^Method Symfony\\(Contracts|Component)\\EventDispatcher\\EventDispatcherInterface::dispatch\(\) invoked with 2 parameters, 1 required\.$#'
+        # PHP 8 Changed the signature of SoapClient::__doRequest()
+        - '#AbusedClient::__doRequest\(\) should be compatible with parameter \$oneWay \(bool\) of method#'
```
Comment (janvernieuwe) [inline phpstan.neon:10]: Thought I added a php 8 check for this?

File: src/Phpro/SoapClient/CodeGenerator/ClientGenerator.php [hunk 2]
```diff
@@ -43,7 +43,8 @@ public function __construct(RuleSetInterface $ruleSet)
     public function generate(FileGenerator $file, $client): string
     {
         try {
-            $class = $file->getClass();
+            // @phpstan-ignore-next-line
```
Comment (janvernieuwe) [inline src/Phpro/SoapClient/CodeGenerator/ClientGenerator.php:46]: ah too bad ;(
Perhaps up our dep requirement?

File: phpstan.neon (hunk 1 above)
Comment (veewee) [reply phpstan.neon:10]: yes, but this one complains about the signature of the overwritten `__doRequest()` function, even though it is valid.
CI complained, so I fixed it :d

File: src/Phpro/SoapClient/CodeGenerator/ClientGenerator.php (hunk 2 above)
Comment (veewee) [reply src/Phpro/SoapClient/CodeGenerator/ClientGenerator.php:46]: laminas-code is an opt-in dependency. We can only up it if we make it a real dependency.

**Review by janvernieuwe — APPROVED** [review]

File: src/Phpro/SoapClient/CodeGenerator/ClientGenerator.php (hunk 2 above)
Comment (janvernieuwe) [reply src/Phpro/SoapClient/CodeGenerator/ClientGenerator.php:46]: or check the version in the dep check?

Comment (veewee) [reply src/Phpro/SoapClient/CodeGenerator/ClientGenerator.php:46]: We could do that...
However: in the long term i'dd love to have a separate "runtime" and a "developer" package.

**Pull request merged**